### PR TITLE
I24 report by user

### PR DESCRIPTION
**Pull request by mate3241**
I24 report by user
I25 report by project

sql query result had to be transformed for frontend, function needs simplifying, possibly rewriting, but it works

### CODE REVIEW

File: backend/seeds/05-example-milestones.js [hunk 1]
```diff
@@ -1,18 +1,20 @@
...
-            name: "Milestone Three",
-            projectId: 3,
-            description: "not a description"
-        }])
+  return knex('milestones').insert([
+      {
+          name: 'milestoneOne',
+          projectId: 1,
+          description: 'milestone 1'
+
```
Comment (ImreTheFifth) [inline backend/seeds/05-example-milestones.js:7]: Do we need this empty line?

File: backend/seeds/05-example-milestones.js [hunk 2]
```diff
@@ -1,18 +1,20 @@
...
+          name: 'milestoneOne',
+          projectId: 1,
+          description: 'milestone 1'
+          
+      },
+      {
+          name: 'milestoneTwo',
+          projectId: 1,
+          description: 'milestone 2'
+
```
Comment (ImreTheFifth) [inline backend/seeds/05-example-milestones.js:13]: Do we need this empty line?

File: backend/seeds/07-example-timeRecords.js [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+            userId: 3,
+            milestoneId: 1,
+            actionLabelId: 1,
+            description: 'something',
+            spentTime: 2,
+            overTime: 1,
+            date: new Date().toISOString().slice(0, 10).replace('T', ' ')
+        },
+        {
+
```
Comment (ImreTheFifth) [inline backend/seeds/07-example-timeRecords.js:13]: empty line

Comment (mate3241) [reply backend/seeds/07-example-timeRecords.js]: sorry

File: backend/src/lib/report.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            }
+        });
+    })
+    calculateAndAddTotal(finalObject);
+    return finalObject;
+}
+
+const getUniqueRowNames = (report, rowName): Array<any> => {
+    let uniqueRowNames = new Set();
+    report.forEach(x => uniqueRowNames.add(x[rowName]));
```
Comment (rastasi) [inline backend/src/lib/report.ts:18]: mi az az x? :)

File: backend/src/lib/report.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    return Array.from(uniqueRowNames);
+}
+
+export const calculateAndAddTotal = (finalObject: object) => {
+    Object.keys(finalObject).forEach(element => {
+        let sum = 0;
+        Object.values(finalObject[element]).forEach(element2 => {
+            sum += +element2;
```
Comment (rastasi) [inline backend/src/lib/report.ts:26]: element és element2?

**Pull request merged**